packages/openlayers/lib/map/handle-errors.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  EndpointError,
+  WfsEndpoint,
+  WmtsEndpoint,
+} from "@camptocamp/ogc-client";
+import { SourceLoadErrorEvent } from "@geospatial-sdk/core";
+import { ImageTile, Map, Tile } from "ol";
+import TileState from "ol/TileState.js";
+
+function handleError(statusCode: number, tile: Tile, map: Map) {
+  tile.setState(TileState.ERROR);
+  map.dispatchEvent(new SourceLoadErrorEvent(statusCode));
+}
+
+export function tileLoadErrorCatchFunction(tile: Tile, src: string, map: Map) {
+  fetch(src).then((response) => {
+    if (response.status === 200) {
```

Comment:
you should check for [`response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) instead, because any HTTP code starting with 2 is a success.